### ISSUE TITLE
Option to disable spinner

### ISSUE THREAD

**Issue by giggio** [microsoft/winget-cli#3494]
### Description of the new feature / enhancement

Add an option to disable the spinner and any other progress indicator.
This adds clutter to the screen if you are trying to capture what happened, and makes it harder to process winget output with command line tools.

### Proposed technical implementation details

Add an option to the settings.json file.

**Comment by denelon** [comment]
You might want to give the "Microsoft.WinGet.Client" PowerShell module or look at the COM API in WinGet for automation scenarios.

**Comment by giggio** [comment]
I'm using the PowerShell module, but it is still incomplete. I need the pins to be available there.
I'll check out the COM API.
Still, one might need to capture the output, for example, in CI scenarios, and it will produce a lot of unnecessary text.

**Comment by giggio** [comment]
I just checked out the API, you can't manage pins through it, either.
https://github.com/microsoft/winget-cli/blob/master/doc/specs/%23888%20-%20Com%20Api.md

**Comment by markusheiliger** [comment]
I agree with @giggio. Getting rid of the spinner output using a command line argument would help to get the captured output clean. BTW, the same applies to the download progress output IMO. TBH I would expect all this output suppressed when using a the /silent switch.

And no, switching to the PowerShell module or using the COM API is no option for me. The automation script works as it is. It's just tooooo noisy!

**Comment by mdanish-kh** [comment]
[Policy] Area-Output

**Comment by jantari** [comment]
The spinners should always be disabled when `--disable-interactivity` parameter is specified by the user.

If interactive features are explicitly disabled, there is absolutely no reason to print spinners, which are a purely interactive feature.

**Comment by asheroto** [comment]
I did something similar but replaced the characters instead of using `Select-String` so that the output could be viewed in real-time...

https://gist.github.com/asheroto/96bcabe428e8ad134ef204573810041f

**Comment by apcsb** [comment]
My solution so far was to look for the separator line and trim everything before it +/- 1 line (in some scripts I need to parse the headers, since winget list, for example, can produce a different number of columns). But you always get the line.

```powershell
$output = WinGet.exe $params
$lineNumber = $output.IndexOf($output -match '----+') - 2
if ($lineNumber -lt 0) { return $null}
return $output[$lineNumber..($output.Length-1)]

```

To make matter worse, when the output is captured in a string, you get encoding issues and skewed columns
Here's an example of `winget list` - check the Inter Graphics Command Center name and id
We need `--disable-interactivity` to properly handle all this

#### Interactive

```
Name                                                        Id                                                               Version            Available    Source
-------------------------------------------------------------------------------------------------------------------------------------------------------------------  
LogiOptionsWordAddin                                        2C559D7A6B8C383AF1FC7ED06D375EAC624F143E6F778C4FCDD50D9E023A4354 10.10.4.0
Tag Explorer                                                46887Dr.Cooper.TagExplorer_s0pf5z833f2vc                         3.7.0.0
Ritt                                                        62283Ritt.Ritt_qr54r3v2h1vqw                                     1.4.0.0
Visual Studio Enterprise 2022                               Microsoft.VisualStudio.2022.Enterprise                           17.9.6                          winget  
LogiOptionsExcelAddin                                       67C3222594B96ED78A8AA3050861B9C717ACC01FF3F90093DB70B6D61707DA57 10.10.4.0
LogiOptionsPowerPointAddin                                  75B10E1044297ED2ACEE0B524BDF62C55A10D7609038BE3D371ABA95D4F4DA6F 10.10.4.0
Intel® Graphics Command Center                              AppUp.IntelGraphicsExperience_8j3eq9eme6ctt                      1.100.5435.0
ThunderboltTM Control Center                                AppUp.ThunderboltControlCenter_8j3eq9eme6ctt                     1.0.37.0
```

#### Captured into a string
```
   \ 
   | 

Name                                    Id                                       Version            Available    Source
-----------------------------------------------------------------------------------------------------------------------
LogiOptionsWordAddin                    2C559D7A6B8C383AF1FC7ED06D375EAC624F143ΓÇª 10.10.4.0
Tag Explorer                            46887Dr.Cooper.TagExplorer_s0pf5z833f2vc 3.7.0.0
Ritt                                    62283Ritt.Ritt_qr54r3v2h1vqw             1.4.0.0
Visual Studio Enterprise 2022           Microsoft.VisualStudio.2022.Enterprise   17.9.6                          winget
LogiOptionsExcelAddin                   67C3222594B96ED78A8AA3050861B9C717ACC01ΓÇª 10.10.4.0
LogiOptionsPowerPointAddin              75B10E1044297ED2ACEE0B524BDF62C55A10D76ΓÇª 10.10.4.0
Intel┬« Graphics Command Center          AppUp.IntelGraphicsExperience_8j3eq9emeΓÇª 1.100.5435.0
ThunderboltTM Control Center            AppUp.ThunderboltControlCenter_8j3eq9emΓÇª 1.0.37.0
```

**Comment by AdreKiseque** [comment]
As far as I can tell, this feature is [already implemented](https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md#visual)? The issue should probably be closed, in that case.

**Comment by giggio** [comment]
> As far as I can tell, this feature is [already implemented](https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md#visual)? The issue should probably be closed, in that case.

@AdreKiseque how so? You linked to config file, that is not an option. I'd only disable progress when piping to another tool, not on every use. Using a global option does not make sense for the provided use case.

**Comment by AdreKiseque** [comment]
Your original issue cited adding an option to the `settings.json` file as a proposed implementation. I only responded in regards to that.
I can see the appeal in having it as a transient option, though.

**Comment by idontsov** [comment]
Is it seriously that hard to add a CLI option that this has been unresolved for years?

**Comment by jantari** [comment]
Agreed, this can't be that hard. While we're waiting on more complex winget issues to be addressed, please just grant us this win

**Comment by God-damnit-all** [comment]
I'm shocked this isn't already an option.

**Comment by youk** [comment]
More whining makes features implemented faster. Well-known fact.

**Comment by cachius** [comment]
Consider this example:
```
PS> $a=(winget list --id Microsoft.WSL)
PS> echo $a

   -
   \


   -
   \
   |

Name                        Id            Version Available Source
------------------------------------------------------------------
Windows Subsystem for Linux Microsoft.WSL 2.6.1.0 2.6.3     winget
```
So the output is riddled with progress indicator output, which makes usage of `Select-Object -First 1`/`head` impossible. To get the info I have to assume progress indicator output is always the same length, which it probably isn't, and use
```
PS> (winget list --id Microsoft.WSL)[9]
Name                        Id            Version Available Source 
```
to get the Available flag. Or use `(winget list --id Microsoft.WSL) -match 'Available'`, but that it works on a dozen lines in the background is a real surprise and overhead. Already had a `--no-progress` hallucinated and the use of `--disable-interactivity` proposed by an LLM - to no effect.

**Comment by denelon** [comment]
It is possible to [disable the spinner in WinGet Settings](https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md#progressbar).

**Comment by giggio** [comment]
> It is possible to [disable the spinner in WinGet Settings](https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md#progressbar).

@denelon I'm aware, this issue touches that (already fixed) and a command line option. See the issue description.

**Comment by denelon** [comment]
@giggio I think we crossed paths with some of the work happening. The PR to add the argument was just merged. I'm expecting the next preview with that feature to get published within a couple of weeks.

**Comment by cachius** [comment]
Thanks @denelon.
For reference, the option to disable via settings.json in `%LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState` or if self-built in  `%LOCALAPPDATA%\Microsoft\WinGet\Settings` is
```
{
    "$schema": "https://aka.ms/winget-settings.schema.json",
    "visual": {
        "progressBar": "disabled"
    }
}
```

**Comment by God-damnit-all** [comment]
> [@giggio](https://github.com/giggio) I think we crossed paths with some of the work happening. The PR to add the argument was just merged. I'm expecting the next preview with that feature to get published within a couple of weeks.

Is there a particular reason that the spinner is tied to stdout in the first place? Utilities like `curl` tend to display progress on stderr so even if you're redirecting stdout you can still see how far along it is.
